web/logging/chargelog/ladelogexportexcel.php

```diff
@@ -0,0 +1,243 @@
+<?php
+
+function makedatetime($start,$f)
+{
+  //echo "$start;$f;";
+  
+   preg_match('/(.*)-(.*)/',$start,$xx);
+   $day=$xx[1];
+   $vonuhr=$xx[2];
+   preg_match('/(.*)-(.*)/',$f,$xx);
+   $bisuhr=$xx[2];
+   
+   echo "$day;$vonuhr - $bisuhr;";
+}
+
+
+
+ if( $_GET['do']=='export')
+ {
+   $dates='';
+   $year=0;
+   $fn='';
+
+   if( isset($_GET['date']) )
+   {
+      $dates=str_replace("-","",$_GET['date']);   
+      $fin="/var/www/html/openWB/web/logging/data/ladelog/$dates.csv";
+      $file=file($fin);
+      asort($file);
+      $fn='Ladelog_'.$dates.'.csv';
+   }
+   else if ( isset($_GET['year']) )
+   {
+      $year=str_replace("-","",$_GET['year']);
+   
+      $files = glob($_SERVER['DOCUMENT_ROOT'] . "/openWB/web/logging/data/ladelog/*.csv");
+      asort($files);
+      $rowClasses = "";
+      $file=[];
+      foreach ($files as $current) 
+      {
+        preg_match('/\/var\/www\/html\/openWB\/web\/logging\/data\/ladelog\/([0-9]{4})([0-9]{2})\.csv/',$current,$m);
+        if( $m[1] == $year )
+        { 
+          $onef = file($current);
+          asort($onef);
+          foreach($onef as $line)
+             if( trim($line) > '')
+               $file[]=$line;
+        }  
+      }
+      $fn='Ladelog_'.$year.'.csv';
+   }
+
+   
+   header('Content-Type: application/csv; charset=UTF-8');
+   header('Content-Disposition: attachment;filename="'.$fn.'";');
+
+   // Haben wir den Kilometerstand vom Ladestart im Ladelog ?   
+   if(file_exists('/var/www/html/openWB/ramdisk/soc1KM'))
+       $head="Tag;Zeit;Km;Kwh;Lade Kw;Ladezeit;Ladepunkt;Lademodus;RFID;Km-Stand\n";
+   else 
+       $head="Tag;Zeit;Km;Kwh;Lade Kw;Ladezeit;Ladepunkt;Lademodus;RFID\n";
+
+  // kopfzeile mit ;
+   echo str_replace(",",";",$head);
+   // daten mit ; und "," als dezimaltrenner
+   foreach($file as $line)
+     {
+       if(trim($line)=='') continue;
+       $fields=explode(",",$line);
+       $idx=0;
+       foreach($fields as $f)
+         {    $idx++;
+              $f=trim($f);
+              switch($idx)
+              {
+               case 1: $starts=$f;
+                       break;
+               case 2: makedatetime($starts,$f);
+                       break;
+               case 6: echo str_replace('H','Std',$f).";";
+                       break;
+               case 8:
+                       switch ((int)$f) 
+                       {
+                        case 0: echo "Sofort;";
+                                break;
+                        case 1: echo "Min+PV;";
+                                break;
+                        case 2: echo "Nur PV;";
+                                break;
+                        case 7: echo "Nachtladen;";
+                                break;
+                        default:
```

Comment:
Bitte die anderen Lademodi der Vollständigkeit halber auch in Klartext übersetzen.

Reply:
Also bitte die "3 = Stop" noch ergänzen.